rpc/client_test.go

```diff
@@ -124,21 +124,21 @@ func TestClientBatchRequest(t *testing.T) {
 	wantResult := []BatchElem{
 		{
 			Method: "test_echo",
-			Args:   []interface{}{"hello", 10, &echoArgs{"world"}},
 			Result: &echoResult{"hello", 10, &echoArgs{"world"}},
 		},
 		{
 			Method: "test_echo",
-			Args:   []interface{}{"hello2", 11, &echoArgs{"world"}},
 			Result: &echoResult{"hello2", 11, &echoArgs{"world"}},
 		},
 		{
 			Method: "no_such_method",
-			Args:   []interface{}{1, 2, 3},
 			Result: new(int),
 			Error:  &jsonError{Code: -32601, Message: "the method no_such_method does not exist/is not available"},
 		},
 	}
+	for i := range batch {
+		batch[i].ArgsOpt = nil
+	}
```

Comment:
Tough decision here... was trying to understand the extent of what we're really trying to test, and decided that dropping the deep equal check for `ArgsOpt` was the easiest but still safe-ish way to go. Something like this is necessary because `ArgsOpt` is a `func` and non-`nil` `funcs` are never deeply equal. There would be other possibilities for testing this but they would require larger changes to this test, so I just stuck with simple for now.